### ISSUE TITLE
alert manager 3.0 alert statuses not populating

### ISSUE THREAD

**Issue by hurculeasz** [alertmanager/alert_manager#256]
Thanks for your work on 3.0. It has many items we need.

When upgrading from 2.2 , the alert statuses are not deployed. same with email templates and notification schemes. I am unable to set statuses on alerts

I cannot find logs that indicate the migration script failed. 

Health Check:
Built-in alert statuses deployed | Failed
-- | --
Default Email Templates deployed | Failed
Default Notification Schemes Deployed | Failed

**Comment by my2ndhead** [comment]
Thanks for reporting this.

- Please add following to $SPLUNK_HOME/etc/apps/alert_manager/local/alert_manager.conf:
`[logging]
rootLevel = DEBUG
logger.alert_manager_migration = DEBUG
`
- Re-enable the migration scripts in $SPLUNK_HOME/etc/apps/alert_manager/local/inputs.conf (use .sh on Unix/.path on Windows):
```
[script://.\bin\alert_manager_migrate-v2.0.sh]
disabled = 0

[script://./bin/alert_manager_migrate-v3.0.sh]
disabled = 0
```
- Check if the health check is green
- Check if $SPLUNK_HOME/var/log/splunk/alert_manager_migration.log contains errors

**Comment by hurculeasz** [comment]
I have added the DEBUG loggging level.

local/inputs.conf did not have v3.0 listed so I have added it. I set disabled = 0 to v2 and v3. I used the perform setup button in Global settings. This did not change the health check status nor the ability to set status. The alert_manager_migration.log did not get updated(hasn't since original 2.2 install)

I am on SPLUNK Version:8.0.2.1

I also attempted moving the alert_manager folder out of apps and loading alert manager 3.0 new. I found that local/inputs did not exist. I created the local/inputs file and added v2 and v3 to this. Same behavior

**Comment by my2ndhead** [comment]
It looks to me that the scripts are not run at all for some reasons.

Can you check in $SPLUNK_HOME/var/log/splunk/splunkd.log if you can find any entries for ExecProcessor?

E.g.
```06-01-2020 20:50:38.153 +0200 INFO  ExecProcessor - New scheduled exec process: /opt/splunk/etc/apps/alert_manager/bin/alert_manager_migrate-v3.0.sh```

**Comment by hurculeasz** [comment]
I do see them get scheduled, then v2 fails due to no module named parse from urllib.parse I do not see subsequent migration script errors.

New scheduled exec process: D:\SPLUNK\bin\Python3.exe D:\SPLUNK\etc\apps\splunk_instrumentation\bin\schedule_delete.py
	cron schedule: "0 0 * * *"
New scheduled exec process: D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_migrate-v2.0.py -index _internal
	interval: run once
New scheduled exec process: D:\SPLUNK\bin\splunk.exe cmd python3.7 D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_migrate-v2.1.py -index _internal
	interval: run once
New scheduled exec process: D:\SPLUNK\bin\splunk.exe cmd python3.7 D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_migrate-v2.2.py -index _internal
	interval: run once
New scheduled exec process: D:\SPLUNK\bin\splunk.exe cmd python3.7 D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_migrate-v3.0.py -index _internal
	interval: run once
New scheduled exec process: D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal
	interval: 60000 ms
New scheduled exec process: D:\SPLUNK/bin/splunkd instrument-resource-usage
	interval: 0 ms
New scheduled exec process: D:\SPLUNK\bin\Python3.exe D:\SPLUNK\etc\apps\splunk_monitoring_console\bin\dmc_config.py
	interval: run once
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_migrate-v2.0.py -index _internal" Traceback (most recent call last):
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_migrate-v2.0.py -index _internal"   File "D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_migrate-v2.0.py", line 4, in <m
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_migrate-v2.0.py -index _internal"     import urllib.parse
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_migrate-v2.0.py -index _internal" ImportError: No module named parse
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal" Traceback (most recent call last):
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal"   File "D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py", line 3, in <module>
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal"     import urllib.parse
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal" ImportError: No module named parse
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal" Traceback (most recent call last):
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal"   File "D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py", line 3, in <module>
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal"     import urllib.parse
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal" ImportError: No module named parse
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal" Traceback (most recent call last):
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal"   File "D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py", line 3, in <module>
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal"     import urllib.parse
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal" ImportError: No module named parse
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal" Traceback (most recent call last):
message from "D:\SPLUNK\bin\splunk.exe cmd python D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py -index _internal"   File "D:\SPLUNK\etc\apps\alert_manager\bin\alert_manager_scheduler.py", line 3, in <module>

**Comment by my2ndhead** [comment]
Thanks! This looks like a Windows plattform issue. Will try to replicate it. Stay tuned...

**Comment by hurculeasz** [comment]
Thank you for looking into this. I have reverted back to a backup of alert_manager 2.2 and the TA app while this is looked at. (I saved the backup of 3.0). Enabling the v2 scripts populated statuses for v2.2. I can rerun the upgrade or switch back to the 3.0 backup when you need testing.

**Comment by my2ndhead** [comment]
I have released a 3.0.1 release on Splunkbase that should fix the issue.

**Comment by hurculeasz** [comment]
I have applied 3.0.1 and the health check passed! Thank you kindly for your help on this.
